src/mirth/elab.mth

```diff
@@ -299,6 +301,14 @@ def(elab-arrow-step-word!, Arrow Word -- Arrow +Elab,
     elab-token@ token-args-0 # high order not yet supported
     elab-word-sig! elab-arrow-apply-sig-type!)
 
+def(elab-arrow-step-stack-assertion!, Arrow Token -- Arrow +Elab,
+    token-brace-arg
+    dip(elab-ctx@)
+    elab-type-stack! drop nip
```

Comment:
I think this is very close. One thing we probably want to check here is that the context you get is the same as the original context `elab-ctx@`. That way you can't introduce new type variables in the assertion by accident. Can do this in a separate PR though -- it's not critical.